lib/smart_proxy_dynflow_core/logger_middleware.rb

```diff
@@ -0,0 +1,28 @@
+module SmartProxyDynflowCore
+  class LoggerMiddleware
+    def initialize(app)
+      @logger = SmartProxyDynflowCore::Log.instance
+      @app = app
+    end
+
+    def call(env)
+      status = 500
+      env['rack.logger'] = @logger
+      @logger.info { "Started #{env['REQUEST_METHOD']} #{env['PATH_INFO']} #{env['QUERY_STRING']}" }
+      @logger.debug { 'Headers: ' + env.select {|k,v| k.start_with? 'HTTP_'}.inspect }
+      @logger.debug { (body = env['rack.input'].read).empty? ? '' : 'Body: ' + body }
+      env['rack.input'].try(:seek, 0) if @logger.debug?
+      before = Time.now.to_f
+      status, _, _ = @app.call(env)
+    rescue Exception => e
+      @logger.exception "Error processing request '#{::Logging.mdc['request']}", e
+      raise e
+    ensure
+      @logger.info do
+        after = Time.now.to_f
+        duration = (after - before) * 1000
```

Comment:
we could check whether the `before` is initialized, in case the exception is raised before line 15